tests/unit/context/thrift_tests.py

```diff
@@ -84,3 +84,49 @@ def test_call_proxy_method(self, mock_enumerate):
         self.assertEqual(self.mock_client.one.call_count, 1)
         self.assertEqual(result, self.mock_client.one.return_value)
         self.assertEqual(self.mock_server_span.make_child.call_args, mock.call("namespace.one"))
+
+    @mock.patch("baseplate.context.thrift._enumerate_service_methods")
+    def test_edge_request_headers(self, mock_enumerate):
+        mock_enumerate.return_value = ["one", "two"]
+        headers = {}
+
+        def mock_set_header(header, val):
+            headers[header] = val
+
+        prot = self.mock_pool.connection().__enter__()
+        prot.trans.set_header = mock_set_header
+        child_span = self.mock_server_span.make_child().__enter__()
+        child_span.context.request_context.header_values.return_value = {
+            'Authentication': 'authn_token',
+            'Edge-Request': 'edge_request_context',
+        }
+
+        proxy = thrift.PooledClientProxy(
+            self.mock_client_cls, self.mock_pool, self.mock_server_span, "namespace")
+        result = proxy.one(mock.sentinel.first, mock.sentinel.second)
+
+        self.assertEqual(headers['Authentication'], 'authn_token')
+        self.assertEqual(headers['Edge-Request'], 'edge_request_context')
+
+    @mock.patch("baseplate.context.thrift._enumerate_service_methods")
+    def test_null_edge_request_headers_not_set(self, mock_enumerate):
+        mock_enumerate.return_value = ["one", "two"]
+        headers = {}
+
+        def mock_set_header(header, val):
+            headers[header] = val
+
+        prot = self.mock_pool.connection().__enter__()
+        prot.trans.set_header = mock_set_header
+        child_span = self.mock_server_span.make_child().__enter__()
+        child_span.context.request_context.header_values.return_value = {
+            'Authentication': None,
+            'Edge-Request': None,
+        }
```

Comment:
nit: you have almost the same chunk here and above, maybe make a helper method ?